### PR TITLE
fix: Safe deep merge typings

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

* Fixes a poorly written assign where functions would be coerced to empty objects if compared - adds a primitive / function check to determine if the value is mergeable - then attempts assigning values.

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Review by dreamwasp — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@5.7.3-alpha.e07022.0<br/>@codecademy/gamut-illustrations@0.17.12-alpha.e07022.0<br/>@codecademy/gamut-kit@0.4.68-alpha.e07022.0<br/>@codecademy/gamut-labs@19.16.6-alpha.e07022.0<br/>@codecademy/gamut-patterns@0.4.7-alpha.e07022.0<br/>@codecademy/gamut-styles@11.6.2-alpha.e07022.0<br/>@codecademy/gamut-tests@2.4.15-alpha.e07022.0<br/>@codecademy/gamut@35.0.1-alpha.e07022.0<br/>@codecademy/konamimojisplosion@0.2.46-alpha.e07022.0<br/>@codecademy/markdown-overrides@0.5.58-alpha.e07022.0<br/>@codecademy/styleguide@42.0.1-alpha.e07022.0<br/>@codecademy/variance@0.17.2-alpha.e07022.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f5b1283d0afb51421a5938--gamut-preview.netlify.app>https://60f5b1283d0afb51421a5938--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f5b1283d0afb51421a5938>Deploy Logs</a></p>

File: packages/variance/src/createTheme/types.ts [hunk 1]
```diff
@@ -17,17 +17,30 @@ export type MergeTheme<
...
+      : B[K]
+    : K extends keyof A
+    ? A[K]
     : never;
 };
 
+/** Extract mergable objects */
+export type Mergable<T> = Exclude<
+  T,
+  (...args: any) => any | string | boolean | symbol | number | any[]
```
[the remark this answers is not in the document]
Comment (codecaaron) [reply packages/variance/src/createTheme/types.ts]: Oops forgot the `()`!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@5.7.3-alpha.511c51.0<br/>@codecademy/gamut-illustrations@0.17.12-alpha.511c51.0<br/>@codecademy/gamut-kit@0.4.68-alpha.511c51.0<br/>@codecademy/gamut-labs@19.16.6-alpha.511c51.0<br/>@codecademy/gamut-patterns@0.4.7-alpha.511c51.0<br/>@codecademy/gamut-styles@11.6.2-alpha.511c51.0<br/>@codecademy/gamut-tests@2.4.15-alpha.511c51.0<br/>@codecademy/gamut@35.0.1-alpha.511c51.0<br/>@codecademy/konamimojisplosion@0.2.46-alpha.511c51.0<br/>@codecademy/markdown-overrides@0.5.58-alpha.511c51.0<br/>@codecademy/styleguide@42.0.1-alpha.511c51.0<br/>@codecademy/variance@0.17.2-alpha.511c51.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f5b728bc7fd19f0d3019f6--gamut-preview.netlify.app>https://60f5b728bc7fd19f0d3019f6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f5b728bc7fd19f0d3019f6>Deploy Logs</a></p>

**Pull request merged**